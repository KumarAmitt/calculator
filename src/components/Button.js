import React from 'react';
import PropTypes from 'prop-types';

const Button = ({
  buttonName, clickHandler, color, wide,
}) => {
  let classNameValue = 'button';
  classNameValue = color === 'primary' ? `${classNameValue} primary` : `${classNameValue} secondary`;
  classNameValue = wide ? `${classNameValue} wide` : classNameValue;

  return (
    <>
      <button type="button" className={classNameValue} onClick={() => clickHandler(buttonName)}>{buttonName}</button>
    </>
  );
};

Button.propTypes = {
  buttonName: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
  color: PropTypes.string.isRequired,
  wide: PropTypes.bool.isRequired,
};

export default Button;
