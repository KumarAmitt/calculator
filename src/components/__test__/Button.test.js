import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Button from '../Button';

test('renders a button correctly', () => {
  const { getByTestId } = render(<Button clickHandler={fireEvent.click} color="orange" wide={false} buttonName="+" />);
  const button = getByTestId('button');

  expect(button.textContent).toBe('+');
});

test('match snapshot of Button', () => {
  const tree = renderer.create(<Button clickHandler={fireEvent.click} color="orange" wide={false} buttonName="+" />).toJSON();

  expect(tree).toMatchSnapshot();
});
