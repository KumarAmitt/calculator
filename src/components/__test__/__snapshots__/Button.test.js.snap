// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot of Button 1`] = `
<button
  className="button secondary"
  data-testid="button"
  onClick={[Function]}
  type="button"
>
  +
</button>
`;
